### ISSUE TITLE
Qwen3-0.6B训练第一个step的梯度为全0

### ISSUE THREAD

**Issue by a31413510** [modelscope/ms-swift#8009]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

使用swift训练Qwen3-0.6B，第一步的grad_norm为inf：
{'loss': 1.67270708, 'grad_norm': inf, 'learning_rate': 1e-05, 'token_acc': 0.78461538, 'epoch': 0.0, 'global_step/max_steps': '1/100', 'percentage': '1.00%', 'elapsed_time': '3s', 'remaining_time': '6m 28s', 'memory(GiB)': 9.01, 'train_speed(iter/s)': 0.254781}
{'loss': 1.89269567, 'grad_norm': 48.32312775, 'learning_rate': 1e-05, 'token_acc': 0.77142857, 'epoch': 0.0, 'global_step/max_steps': '2/100', 'percentage': '2.00%', 'elapsed_time': '7s', 'remaining_time': '5m 45s', 'memory(GiB)': 9.59, 'train_speed(iter/s)': 0.283668}
进一步打印梯度，发现第一步的梯度全是0

<img width="960" height="609" alt="Image" src="https://github.com/user-attachments/assets/cda519d5-0e17-458c-a14e-bdf7a0a2e0d4" />

### How to Reproduce / 如何复现

训练命令：
`
CUDA_VISIBLE_DEVICES=0 \
swift sft \
    --model /work/Qwen3-0.6B-Base \
    --train_type full \
    --torch_dtype float32 \
    --dataset '/work/datasets/gsm8k_messages.jsonl' \
    --use_chat_template true \
    --model_type qwen3 \
    --packing false \
    --padding_free false \
    --dataset_shuffle false \
    --train_dataloader_shuffle false \
    --dataloader_num_workers 1 \
    --dataset_num_proc 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --max_length 4096 \
    --learning_rate 1e-5 \
    --lr_scheduler_type constant\
    --gradient_accumulation_steps 1 \
    --warmup_ratio 0.00 \
    --num_train_epochs 1 \
    --max_steps 1 \
    --eval_steps 50000 \
    --save_steps 50000 \
    --save_total_limit 1 \
    --logging_steps 1 \
    --output_dir output \
    --use_logits_to_keep false \
    --seed 42 
`

### Additional Information / 补充信息

环境信息
ms-swift                  release/3.12
transformers           4.57.6
torch                        2.9.1

**Comment by Jintao-Huang** [comment]
加个这个 --fp16 false
